src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -559,13 +561,23 @@ public void createSnapshotWithProvidedIds(
 
             List<String> rowIds = table.getRowIds();
             if (rowIds.size() > 0) {
-                ST sqlTemplate = new ST(loadRootRowIdsTemplate);
-                sqlTemplate.add("project", projectId);
-                sqlTemplate.add("snapshot", snapshotName);
-                sqlTemplate.add("dataset", datasetBqDatasetName);
-                sqlTemplate.add("tableId", sourceTable.getId().toString());
-                sqlTemplate.add("rowIds", rowIds);
-                bigQueryProject.query(sqlTemplate.render());
+
+                int maxValuesInArray = 10000; // To prevent BQ choking on a huge array, split it up into chunks
+                long iterationsToLoopThrough = round(ceil(rowIds.size() / maxValuesInArray));
+
+                for (int i = 0; i <= iterationsToLoopThrough; i++) {
+                    int start = maxValuesInArray * i;
+                    int end = (rowIds.size() - start) < maxValuesInArray ? rowIds.size() : start + maxValuesInArray;
```

Comment:
That was my first reaction too--but the subList method actually is inclusive at the start and exclusive at the end. So we want to go all the way from 0 to the originalList.size(). This could use a comment as well.

Reply:
Oh! Ok, my bad! That looks good as is, then!

Reply:
Do you think this would be way clearer if i wrote a helper method that took start and end as inputs and the remainder was a separate call before the start of the loop?